src/configure

```diff
@@ -6135,8 +6135,58 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
 
-   warn="-Wstringop-truncation"
- nowarn="-Wno-stringop-truncation"
+   if test "x$werror" = xyes; then :
+  if test 1 -gt 0; then :
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC supports -Werror" >&5
+$as_echo_n "checking if $CC supports -Werror... " >&6; }
+fi
+  ac_ext=c
+ac_cpp='$CPP $CPPFLAGS'
+ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_compiler_gnu=$ac_cv_c_compiler_gnu
+
+  ac_saved_cflags="$CFLAGS"
+  CFLAGS="-Werror -Werror"
```

Comment:
The repeated `-Werror -Werror` looks weird.

Reply:
Right, lulz, apparently we're using `-Werror` to test if `-Werror` is supported 🤣 

It's a peculiar result of the "compiled" macros, but I'm not going to touch it as long as everything seems to work (also it's unrelated to this PR).